addon-test-support/index.js

```diff
@@ -0,0 +1,58 @@
+import { get } from '@ember/object';
+import { getContext, settled } from '@ember/test-helpers';
+import Test from 'ember-simple-auth/authenticators/test';
+
+const SESSION_SERVICE_KEY = 'service:session';
+const TEST_CONTAINER_KEY = 'authenticator:test';
+
+function ensureAuthenticator() {
+  const { owner } = getContext();
+  const authenticator = owner.lookup(TEST_CONTAINER_KEY);
+  if (!authenticator) {
+    owner.register(TEST_CONTAINER_KEY, Test);
+  }
+}
+
+/**
+ * Authenticates the session.
+ *
+ * @param {Object} sessionData Optional argument used to mock an authenticator
+ * response (e.g. a token or user).
+ * @return {Promise}
+ * @public
+ */
+export function authenticateSession(sessionData) {
+  const { owner } = getContext();
+  const session = owner.lookup(SESSION_SERVICE_KEY);
+  owner.setupRouter(); // router must initialize fully before authentication
+  ensureAuthenticator();
```

Comment:
If this function is private, you could just pass the owner in.

Reply:
I'm not sure I understand what you mean

Reply:
Oh wait, gotcha. So the getContext isn't looked up twice. I'll do that